pkg/controller/backup_status_updater.go

```diff
@@ -100,30 +113,219 @@ func (u *realBackupConditionUpdater) Update(backup *v1alpha1.Backup, condition *
+	}
+
+	switch condition.Command {
+	case v1alpha1.LogStartCommand:
+		// start command should not update TimeCompleted and condition when start complete.
+		// start command other status and condition info can be synced to the whole log backup status.
+		if condition.Type == v1alpha1.BackupComplete {
+			if copyStatusP != nil {
+				copyStatusP.TimeCompleted = nil
+			}
+			copyConditionP = nil
+		}
+	case v1alpha1.LogStopCommand:
+		// stop command should not update TimeStarted, but should update condition when stop complete.
+		// stop command other status should not be synced to whole log backup status.
+		if condition.Type == v1alpha1.BackupComplete {
+			if copyStatusP != nil {
+				copyStatusP.TimeStarted = nil
+			}
+		} else {
+			copyStatusP = nil
+			copyConditionP = nil
+		}
+	case v1alpha1.LogTruncateCommand:
+		// truncate command shoudld not update TimeStarted, TimeCompleted when truncate complete.
+		// truncate command other status should not be synced to whole log backup status.
+		if condition.Type == v1alpha1.BackupComplete {
+			if copyStatusP != nil {
+				copyStatusP.TimeCompleted = nil
+				copyStatusP.TimeStarted = nil
+			}
+		} else {
+			copyStatusP = nil
+		}
+		copyConditionP = nil
+	default:
+		// should not hanpen
+		return false
+	}
+
+	// update whole log status and condition
+	wholeStatusUpdate := updateBackupStatus(&backup.Status, copyStatusP)
+	wholeConditionUpdate := v1alpha1.UpdateBackupCondition(&backup.Status, copyConditionP)
+
+	return wholeStatusUpdate || wholeConditionUpdate
+}
+
+// updateLogSubCommandStatusOnly only updates log subcommand's status info.
+func updateLogSubCommandStatusOnly(status *v1alpha1.LogSubCommandStatus, newStatus *BackupUpdateStatus) bool {
+	isUpdate := false
+	if newStatus == nil {
+		return isUpdate
+	}
+	if newStatus.TimeStarted != nil && status.TimeStarted != *newStatus.TimeStarted {
+		status.TimeStarted = *newStatus.TimeStarted
+		isUpdate = true
+	}
+	if newStatus.TimeCompleted != nil && status.TimeCompleted != *newStatus.TimeCompleted {
+		status.TimeCompleted = *newStatus.TimeCompleted
+		isUpdate = true
+	}
+	if newStatus.LogTruncatingUntil != nil && status.LogTruncatingUntil != *newStatus.LogTruncatingUntil {
+		status.LogTruncatingUntil = *newStatus.LogTruncatingUntil
+		isUpdate = true
+	}
+	return isUpdate
+}
+
+// updateLogSubCommandConditionOnly only updates log subcommand's condition info.
+func updateLogSubCommandConditionOnly(status *v1alpha1.LogSubCommandStatus, condition *v1alpha1.BackupCondition) bool {
+	isUpdate := false
+	if condition == nil {
+		return isUpdate
+	}
+
+	if status.Phase != condition.Type {
+		status.Phase = condition.Type
+		isUpdate = true
+	}
+
+	var oldCondition *v1alpha1.BackupCondition
+	// find old Backup condition.
+	index := -1
+	for i, c := range status.Conditions {
+		if c.Type == condition.Type {
+			index = i
+			break
+		}
+	}
+	if index != -1 {
+		oldCondition = &status.Conditions[index]
+	} else {
```

Comment:
if old condition not exist, why not just make the update and return

Reply:
ok